freedomofpress/securedrop
# Adds CI pipeline for testing staging environment on AWS

## Status

Ready for review :+1:

## Description of Changes

Fixes 
* #1067 -- AUTOMATIC testing of staging in CI utilizing CircleCI. Development will still be run under travis, I'd like to break off a new ticket to also run development in CircleCI in parallel.
* #1146 -- So I started to tackle this a bit. I'm using `ansible 2.2.1.0` during CI to deploy securedrop. This is getting used in staging and development testing in CI (in travis and staging) and passing. Had to make very minimal tweaks to get working... Was kind of surprised. There are some other legacy attributes we are using that I want to do a full cleanup on. I'll open a separate PR for that as part of the development under CircleCI effort.
* Lots of little string comparison fixes in the `testinfra` tests.

Changes
* Refactors the `build-generic-pkg` role to work against remote machines, this was necessary to get CI working properly. Previous role and playbook logic assumed that securedrop files were being mounted via virtual-machine shared storage mount.

## Testing

How should the reviewer test this PR?

You should really test locally as well as test trigger the CI. If you want to test locally, you'll first off need AWS creds. You'll also need the following environment variables set:
* `BUILD_NUM` - arbitrary string, just dont conflict with anyone else on the team or MORE importantly the build number that is running in CI. 
* `FPF_CI` - set to `true`
* `FPF_GRSEC` - set to `false` to skip grsec tests
* `CI_AWS_VPC_ID`
* `CI_AWS_REGION` - currently `us-west-1`
* `CI_AWS_TYPE` -- been doing `m3.medium` so far

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.

This is going to break current users playbook workflow without further fleshed out documentation. It makes some changes that are incompatible with the previously utilized version of `ansible 1.7.4` so they'll need to run a current version of ansible under a virtualenv under tails. 

2. New installs.

Need to clean up documentation and write additional code (recommend we do this in #1146 ) to bootstrap the provisioning using the more modern version of ansible than what is supplied in the tails repos.

## Checklist

### If you made changes to the system configuration:

- [x] Testinfra tests pass on the staging VM
